python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -70,6 +90,7 @@ def inputs_done(self) -> None:
         super().inputs_done()
 
     def has_next(self) -> bool:
+        assert self._started
```

Comment:
should we add the assertion to `get_next()` as well?

Reply:
btw do we plan to update `InputDataBuffer` in a separate PR?

Reply:
Sure. Yes, we should update it separately.